Formula/pnpm.rb

```diff
@@ -20,11 +20,63 @@ class Pnpm < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "efcac832dc165a64bbb93b7018876cf7dc70d669ae3b1f2a68c46ea0bcddc572"
   end
 
-  depends_on "node"
+  depends_on "node" => :build
+
+  resource "pnpm-buildtime" do
+    url "https://get.pnpm.io/v6.7.js"
+    sha256 "56bffe269aab055bec3a07fa65c387f5a35729550dc7fc76711844a2de6e1b8b"
+  end
 
   def install
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
+    resource("pnpm-buildtime").stage do |r|
+      (buildpath/"buildtime-bin").install "v#{r.version}.js"
+      IO.write "pnpm", <<~EOS
+        #!/bin/sh
+        node #{buildpath}/buildtime-bin/v#{r.version}.js "$@"
+        exit "$?"
+      EOS
+      chmod 0755, "pnpm"
+      (buildpath/"buildtime-bin").install "pnpm"
+    end
+    ENV.prepend_path "PATH", buildpath/"buildtime-bin"
+    system "pnpm", "install"
+    system "pnpm", "run", "compile-only"
+    on_macos do
+      if Hardware::CPU.arm?
+        chdir "packages/pnpm" do
+          mkdir_p "../artifacts/macos-arm64"
+          system "node_modules/.bin/pkg", "./dist/pnpm.cjs",
+            "--out-path=../artifacts/macos-arm64", "--targets=node14-macos-arm64"
```

Comment:
I'm not a fan of hardcoding `node14` here.

Reply:
I will ask upstream to support ARM64 binary.
I will remove the ARM64-related definition until I hear from upstream.